src/game/logic/PlayerLogic.java

```diff
@@ -41,4 +41,21 @@ public boolean isPhaseOne(){
     }
 
     public PlayerToken[] getBoardAsPlayerTokens(){ return gameBoard.getBoard(); }
-}
\ No newline at end of file
+
+    public boolean removePiece(int index) {
+        if (activePlayer.getPlayerToken() == gameBoard.getCell(index).getPlayer()) { // i wrote this
```

Comment:
Nit: Comment not needed

Reply:
its a self reference as I wrote the last part of the first story